pkg/log/log.go

```diff
@@ -70,6 +74,16 @@ func SetOutput(sink OutputSink) error {
 	}
 }
 
+func addStackTraceField(e *logrus.Entry, err error) *logrus.Entry {
+	if e != nil && err != nil {
+		split := strings.SplitAfter(fmt.Sprintf("%+v", err), err.Error())
```

Comment:
This method of logging the stack trace presumes that the error was created with `github.com/pkg/errors.New(()` or wrapped with `github.com/pkg/errors.Wrap()`.  This will not be true for kio/errors.  That's likely OK for now.  

Given that dependence though, a better way to check for the stacktrace is to use a typecast to errors.stackTracer interface (Apparently that works even if the interface type is not exported: https://godoc.org/github.com/pkg/errors).  Otherwise use a local definition of an interace and check for presence of StackTrace() method.  It's still OK to use the fmt.Sprintf("%+v") since the errors docs document it.

Reply:
So, the reason why I took this approach was because I tested it with both error created by kio/errors and github.com/pkg/errors. 
For kio/error:  fmt.Sprintf("%+v", err) EQUALS err.Error() and there is no stack trace. So len(split) will always be 1 for this case.
For github.com/pkg/errors: there is no guarantee that we get a stackTrace for each error, so  len(split) > 1  only when we have a stackTrace.